### PR TITLE
pkg/transform: rename to metricfamily, separate source files

### PR DESCRIPTION
**Pull request by s-urbaniak**
This is just a mechanical refactor for better readability.

### CODE REVIEW

File: pkg/metricfamily/sort.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+	clientmodel "github.com/prometheus/client_model/go"
+)
+
+func SortMetrics(family *clientmodel.MetricFamily) (bool, error) {
+	sort.Sort(MetricsByTimestamp(family.Metric))
+	return true, nil
+}
+
+type MetricsByTimestamp []*clientmodel.Metric
```
Comment (metalmatze) [inline pkg/metricfamily/sort.go]: Do we really need to use these "old" interfaces? I think you can simply use the `sort.Slice` func with the `func (m MetricsByTimestamp) Less(i int, j int) bool` as parameter. 😊

File: pkg/metricfamily/sort.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		dst.Metric = result
+		families[pos] = nil
+	}
+	return Pack(families)
+}
+
+// PackedFamilyWithTimestampsByName sorts a packed slice of metrics
+// (no nils, all families have at least one metric, and all metrics
+// have timestamps) in order of metric name and then oldest sample
+type PackedFamilyWithTimestampsByName []*clientmodel.MetricFamily
```
Comment (metalmatze) [inline pkg/metricfamily/sort.go]: I can't find a reference of this in the diff anymore. 🤔

Comment (s-urbaniak) [reply pkg/metricfamily/sort.go]: It was here: https://github.com/openshift/telemeter/blob/676f4b315e50419457013d8180a6bc20f8be74db/pkg/transform/transform.go#L39

File: pkg/metricfamily/sort.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/metricfamily/sort.go]: Agreed :+1: This is just a first mechanical refactor (hopefully) without actual logic changes.

Comment (squat) [reply pkg/metricfamily/sort.go]: right! logic and opinionated clean up will follow shortly!

**Pull request merged**